### PR TITLE
Export patches as C program source

### PR DESCRIPTION
**Pull request by torusrxxx**
#686.
Template is implemented, but is embedded in the source of x64dbg in the form of raw string now. If someone comes about a new template, then maybe we have to load them dynamically from file.

---
This change is [Reviewable](https://reviewable.io/reviews/x64dbg/x64dbg/712)

### CODE REVIEW

File: src/gui/Src/Gui/PatchDialog.cpp [hunk 1]
```diff
@@ -682,8 +684,228 @@ void PatchDialog::saveAs1337(const QString & filename)
...
+    msg.setParent(this, Qt::Dialog);
+    msg.setWindowFlags(msg.windowFlags() & (~Qt::WindowContextHelpButtonHint));
+    msg.exec();
+}
+
+void PatchDialog::saveAsC(const QString &filename)
+{
+    QString template_C;
+    // Loading template
+    template_C = QString(R"xxx(
```
Comment (mrexodia) [inline src/gui/Src/Gui/PatchDialog.cpp]: Is this supported by Visual Studio 2013? My preference would be to load templates from a file that is not contained in the x64dbg source code directly (we can create a separate repository for it and include it as a submodule). I also think there should be a separate button for saving with a template so the current behavior is unchanged.

**Comment by torusrxxx** [comment]
I have built it successfully.VS2013 professional.

**Comment by mrexodia** [comment]
Sorry, but I don't want to accept the pull request in its current form. I will check it out and see how it will fit in.

**Comment by mrexodia** [comment]
This is till on my todo list, will try to find time for it.

**Comment by mrexodia** [comment]
Implemented as a plugin: https://github.com/torusrxxx/x64dbgpatchexporter

**Pull request closed without merging**